### PR TITLE
add LookupJoin and LookupJoinTest changes from summingbird

### PR DESCRIPTION
**Pull request by adamkozuch**

### CODE REVIEW

**Comment by adamkozuch** [comment]
Issue #1121

File: scalding-core/src/main/scala/com/twitter/scalding/typed/LookupJoin.scala [hunk 1]
```diff
@@ -38,7 +58,7 @@ import java.io.Serializable
  * The entries in the left pipe's tuples have the following
  * meaning:
  *
- * T: The time at which the (K, W) lookup occurred.
+ * T: The the time at which the (K, W) lookup occurred.
```
Comment (johnynek) [inline scalding-core/src/main/scala/com/twitter/scalding/typed/LookupJoin.scala]: should probably fix this. Looks like someone updated the comments.

File: scalding-core/src/main/scala/com/twitter/scalding/typed/LookupJoin.scala [hunk 2]
```diff
@@ -30,7 +50,7 @@ import java.io.Serializable
  *
  * To simulate this behavior, lookupJoin accepts pipes of key-value
  * pairs with an explicit time value T attached. T must have some
- * sensible ordering. The semantics are, if one were to hit the
+ * sensical ordering. The semantics are, if one were to hit the
```
Comment (johnynek) [inline scalding-core/src/main/scala/com/twitter/scalding/typed/LookupJoin.scala]: my joke word, sensical, was removed as it is nonsensical.

File: scalding-core/src/main/scala/com/twitter/scalding/typed/LookupJoin.scala [hunk 3]
```diff
@@ -75,12 +135,15 @@ object LookupJoin extends Serializable {
...
       left.map { case (t, (k, v)) => (k, (t, Left(v): Either[V, JoinedV])) }
-        .++(right.map { case (t, (k, joinedV)) => (k, (t, Right(joinedV): Either[V, JoinedV])) })
+        .++(right.map {
+        case (t, (k, joinedV)) =>
+          (k, (t, Right(joinedV): Either[V, JoinedV]))
+      })
         .group
         .withReducers(reducers.getOrElse(-1)) // -1 means default in scalding
-        .sortBy(identity) // time then left before right
+        .sortBy(identity)
```
Comment (johnynek) [inline scalding-core/src/main/scala/com/twitter/scalding/typed/LookupJoin.scala]: this should just be .sorted, sortBy(identity) is indirect.

**Comment by johnynek** [comment]
Thanks for doing this.

**Comment by adamkozuch** [comment]
I made correction

**Comment by johnynek** [comment]
Thank you sir!

**Pull request merged**